src/plugins/intel_gna/backend/make_pwl.cpp

```diff
@@ -363,51 +334,100 @@ void make_gna_pwl(const DnnActivation&  fun,
             break;
         }
-        case kActPow: {
-            float pow_exponent = fun.args.pow.exponent;
-            IE_ASSERT(pow_exponent != 1.0f);
-            if (pow_exponent == 0.0f) {
-                int32_t x_lower = INT32_MIN;
-                int32_t x_upper = INT32_MAX;
-                int16_t y_lower = FLOAT_TO_INT16(1 * out_scale);
-                int16_t y_upper = y_lower;
-                auto n_segments = 2;
-
-                gna_pwl.resize(n_segments);
-
-                gna_pwl[0].xBase = INT32_MIN & XBASEMASK;  // zero out the 2 lsb
-                gna_pwl[0].yBase = y_lower;
-                gna_pwl[0].slope = 0;
-                print_segment(gna_pwl[0].xBase / in_scale, gna_pwl[0].yBase / out_scale, 0.0);
-
-                gna_pwl[1].xBase = x_lower & XBASEMASK;  // zero out the 2 lsb
-                gna_pwl[1].yBase = y_lower;
-                double slope = (static_cast<double>(y_upper - y_lower) / out_scale) / (static_cast<double>(x_upper - x_lower) / in_scale);
-                s = gna_slope(slope, in_scale, out_scale);
-                gna_pwl[1].slope = FLOAT_TO_INT16(s.slope * s.slope_scale);
-                gna_pwl[1].xBase = gna_pwl[1].xBase | s.slope_scale_index;
-                print_segment((int32_t)(gna_pwl[1].xBase & XBASEMASK) / in_scale, gna_pwl[1].yBase / out_scale, 1.0);
-                if (INT32_MAX > x_upper) {  // need a right segment
-                    gna_pwl.push_back({
-                        static_cast<int32_t>(x_upper & XBASEMASK),  // zero out the 2 lsb
-                        y_upper,
-                        0 });
-
-                    print_segment((x_upper & XBASEMASK) / in_scale, gna_pwl[2].yBase / out_scale, 0.0);
-                }
-            } else {
-                double min_x_val = -pwl[0].b / pwl[0].m;
-                double max_x_val = (y_max/out_scale - pwl[pwl_size - 2].b) / pwl[pwl_size - 2].m;
-                double min_y_val = fun.fqParams.set ? pwl[0].beta : 0;
-                double max_y_val = fun.fqParams.set ? pwl.front().beta : y_max / out_scale;
-                gna_pwl = create_multisegment_gna_pwl(pwl, in_scale, out_scale, min_x_val, max_x_val, min_y_val, max_y_val,
-                    fun.fqParams.set, true);
-                break;
-            }
-            break;
-        }
         default:
             THROW_GNA_EXCEPTION << "Unexpected function activation!" << fun;
     }
     insert_extra_pwl_segments(gna_pwl, y_min, y_max);
 }
+
+template<typename T>
+static void make_gna_pwl(const T* m,
+                         const T* b,
+                         const T* alpha,
+                         size_t count,
+                         double in_scale,
+                         double out_scale,
+                         std::vector<gna_pwl_segment_t> &gna_pwl) {
+    gna_pwl.clear();
+    for (size_t i = 0; i < count; i++) {
```

Comment:
are segments created for values which are outside the range for which pwl search was run (less than the first segment xbase or greater than the maximum value)? What will be a result if x is outside this range?

Reply:
Two segments are added that cover cases when value of pwl search is out of range.